src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -1785,17 +1787,26 @@ public void startVirtualMachine(AzureVMAgent agent) throws AzureCloudException {
         LOGGER.log(Level.INFO, "AzureVMManagementServiceDelegate: startVirtualMachine: {0}", agent.getNodeName());
         int retryCount = 0;
         boolean successful = false;
+        boolean isStartTimeout = false;
 
         while (!successful) {
             try {
-                azureClient.virtualMachines().getByResourceGroup(agent.getResourceGroupName(), agent.getNodeName()).start();
-                successful = true; // may be we can just return
+                VirtualMachine virtualMachine = azureClient.virtualMachines().getByResourceGroup(agent.getResourceGroupName(), agent.getNodeName());
+                if (isStartTimeout) {
```

Comment:
Can we use `retryCount == 0` to handle the conditional check?

Reply:
You mean something like when the VM is started successfully, set the `retryCount` as 0, right?